Segment-Optimizely-X/Classes/SEGOptimizelyXIntegration.m

```diff
@@ -159,13 +162,16 @@ - (void)setupTimer
 
 - (void)flushQueue:(NSMutableArray *)queue
 {
-    for (SEGTrackPayload *obj in queue) {
-        [self track:obj];
-        SEGLog(@"Calling track with payload:%@", obj);
-    }
+    @synchronized(queue)
```

Comment:
I'm confused. I thought we are using a dispatch queue, why is `@synchronized` directive being used?